node/src/components.rs

```diff
@@ -206,3 +206,32 @@ pub(crate) trait ValidatorBoundComponent<REv>: Component<REv> {
         rng: &mut NodeRng,
     ) -> Effects<Self::Event>;
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Since this module isn't `pub` or `pub(crate)`, nor is anything inside it re-exported with `pub use`, is it even accessible?

Also, nitpick: it's a bit weird to me that a module is called `tests` and it contains no tests :P Maybe `test_utils`?

Reply:
It's there to provide access to `TransactionCategory` for both Block Validator tests and Transaction Buffer tests. Hence I put it in `components.rs` as this is the common denominator for these two.

Reply:
Updated to a non-mod version in 98665645b40af712399d47253b7765c57ea66326